clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcsImageProvider.java

```diff
@@ -23,119 +23,85 @@
 import java.util.ArrayList;
-import java.util.Arrays;
+import java.util.Collections;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
-@RestController
-@RequestMapping("/ecs")
-public class EcsFindImagesByTagController {
+@Component
+public class EcsImageProvider implements ImageRepositoryProvider {
   private static final Pattern ACCOUNT_ID_PATTERN = Pattern.compile("^([0-9]{12})");
   private static final Pattern REPOSITORY_NAME_PATTERN = Pattern.compile("\\/([a-z0-9._-]+)");
   private static final String IDENTIFIER_PATTERN = "(:([a-z0-9._-]+)|@(sha256:[0-9a-f]{64}))";
   private static final Pattern REGION_PATTERN = Pattern.compile("(\\w+-\\w+-\\d+)");
-  private static final Pattern REPOSITORY_URI_PATTERN = Pattern.compile(ACCOUNT_ID_PATTERN.toString() + ".+" +
+  private static final Pattern REPOSITORY_URI_PATTERN = Pattern.compile(ACCOUNT_ID_PATTERN.toString() + "\\.dkr\\.ecr\\." +
     REGION_PATTERN.toString() + ".+" +
     REPOSITORY_NAME_PATTERN.toString() +
     IDENTIFIER_PATTERN);
 
-  AmazonClientProvider amazonClientProvider;
+  private final AmazonClientProvider amazonClientProvider;
 
-  AccountCredentialsProvider accountCredentialsProvider;
+  private final AccountCredentialsProvider accountCredentialsProvider;
 
   @Autowired
-  public EcsFindImagesByTagController(AmazonClientProvider amazonClientProvider,
-                                      AccountCredentialsProvider accountCredentialsProvider) {
+  public EcsImageProvider(AmazonClientProvider amazonClientProvider,
+                          AccountCredentialsProvider accountCredentialsProvider) {
     this.amazonClientProvider = amazonClientProvider;
     this.accountCredentialsProvider = accountCredentialsProvider;
   }
 
-  private NetflixAmazonCredentials getCredentials(String accountId) {
-    for (AccountCredentials credentials : accountCredentialsProvider.getAll()) {
-      if (credentials instanceof NetflixAmazonCredentials) {
-        NetflixAmazonCredentials amazonCredentials = (NetflixAmazonCredentials) credentials;
-        if (amazonCredentials.getAccountId().equals(accountId)) {
-          return amazonCredentials;
-        }
-      }
-    }
-
-
-    throw new NotFoundException(String.format("AWS account %s was not found.  Please specify a valid account name"));
+  @Override
+  public String getRepositoryName() {
+    return "ECR";
   }
 
-  @RequestMapping(value = "/images/find", method = RequestMethod.GET)
-  public Object findImage(@RequestParam("q") String dockerImageUrl, HttpServletRequest request) {
-    // TODO: Make this method repo-generic, to allow support for say DockerHub. Have a method findEcrImage that contains the code below, and a method such as findDockerHubImage for when DockerHub repo is supported.
+  @Override
+  public boolean handles(String url) {
+    return url.contains(".dkr.ecr.");
```

Comment:
I'm fine with this definition for now, but a more complex check might be safer in the future

Reply:
There's a `REPOSITORY_URI_PATTERN` that can be used.

Though, using it here will mean that the later check:
```
    if (!isValidEcrUrl(url)) {
      throw new Error("The repository URI provided is not properly structured.");
    }
```
will be pointless.

Reply:
Let's cross the bridge once we get to the river

Reply:
I've switched it to use the pattern, and removed that if-statement from later on.

Doesn't make sense to say "Yeah this URI is supported by ECR - But, wait, it's malformed/not properly structured." 

Supported should mean it's all good, unsupported should mean it's not.